mirrord/layer/tests/apps/dns_resolve/src/main.rs

```diff
@@ -0,0 +1,47 @@
+use std::{ffi::CString, ptr, str};
+
+fn get_ip_addresses(node: &str) -> (Vec<String>, i32) {
+    let node = CString::new(node).unwrap();
+    let service = CString::new("http").unwrap();
+
+    let mut hints: libc::addrinfo = unsafe { std::mem::zeroed() };
+    hints.ai_family = libc::AF_UNSPEC;
```

Comment:
not sure why you do it, passing empty hints is ok?

Reply:
I am not sure if converting to a readable form like so is needed:
 ```rs
let ipv4_addr = std::net::Ipv4Addr::from(ip);
let addr_str = ipv4_addr.to_string();

addresses.push(addr_str);
```
I wanted to make sure that the addresses we get are not empty, so maybe I should just push `(*sockaddr_in).sin_addr.s_addr` without converting it to a readable form? 

Are we concerned about asserting the response is not empty or just the return code?